test/upgrade/fleet_test.go

```diff
@@ -53,7 +53,7 @@ func NewFleet(t *testing.T, version string) *Fleet {
 	//nolint:gosec // does not need to be secure for tests
 	projectName := "fleet-test-" + strconv.FormatUint(rand.Uint64(), 16)
 
-	dockerClient, err := client.NewClientWithOpts(client.FromEnv)
+	dockerClient, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
```

Comment:
I pushed this which allows the newer client to work with older versions of Docker Desktop. This fixes the issue that we addressed with https://github.com/fleetdm/fleet/pull/10570.